### PR TITLE
dev-util/phpstorm: new package

### PR DESCRIPTION
**Pull request by switch87**
new ebuild for jetbrains PhpStorm
https://bugs.gentoo.org/show_bug.cgi?id=563120

### CODE REVIEW

**Comment by hasufell** [comment]
the first commit message should rather be named
```
licenses: add PhpStorm_Academic_license for dev-utils/phpstorm
```

**Comment by hasufell** [comment]
typo in second commit message... there is no category `dev-utils`, you mean `dev-util`

**Comment by switch87** [comment]
:+1:

File: dev-util/phpstorm/phpstorm-9.0.2.141.2462.ebuild [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+SLOT="$(get_major_version)"
+MY_PV="$(get_version_component_range 4-5)"
+MY_PN="PhpStorm"
+MY_PN2="webide"
+
+DESCRIPTION="hpStorm’s smart code editor provides excellent support for PHP (including the latest language versions and frameworks), HTML, JavaScript, CSS, Sass, Less, CoffeeScript, and many other languages. Enjoy coding with intelligent context-aware code completion, error detection, and on-the-fly code inspections & fixes."
+HOMEPAGE="http://www.jetbrains.com/phpstorm"
+SRC_URI="http://download.jetbrains.com/${MY_PN2}/${MY_PN}-$(get_version_component_range 1-3).tar.gz"
+
+LICENSE="PhpStorm_license PhpStorm_Academic_license PhpStorm_Classroom_license PhpStorm_OpenSource_license PhpStorm_personal_license"
```
Comment (mgorny) [inline dev-util/phpstorm/phpstorm-9.0.2.141.2462.ebuild]: Not sure if some of them shouldn't use `|| ()`. Do you need to read all of them to use this? ;-P

Comment (switch87) [inline dev-util/phpstorm/phpstorm-9.0.2.141.2462.ebuild]: Oke :+1:  I did not know about the option.

**Comment by mgorny** [comment]
@switch87, if I may ask, did you copy those license files from somewhere or 'create' them from some source yourself? Because github indicates there's no newline at end of file which sucks a bit for Linux ;-).

Lemme try to find some local PHP expert to look at this too.

File: dev-util/phpstorm/phpstorm-9.0.2.141.2462.ebuild [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+# Distributed under the terms of the GNU General Public License v2
+# $Id$
+
+EAPI=5
+inherit eutils versionator
+
+SLOT="$(get_major_version)"
+MY_PV="$(get_version_component_range 4-5)"
+MY_PN="PhpStorm"
+MY_PN2="webide"
```
Comment (mgorny) [inline dev-util/phpstorm/phpstorm-9.0.2.141.2462.ebuild]: Now I'm being picky but please don't abuse variables too much ;-). You don't use this one anywhere else, so better just inline it in SRC_URI.

File: dev-util/phpstorm/phpstorm-9.0.2.141.2462.ebuild [hunk 3]
```diff
@@ -0,0 +1,33 @@
+# Copyright 1999-2015 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Id$
+
+EAPI=5
+inherit eutils versionator
+
+SLOT="$(get_major_version)"
+MY_PV="$(get_version_component_range 4-5)"
```
Comment (mgorny) [inline dev-util/phpstorm/phpstorm-9.0.2.141.2462.ebuild]: This is also used in one place only. Better to inline it since MY_PV suggests it's about the filename actually.

Or even better, you could drop the two extra version components from filename, and inherit `vcs-snapshot` to handle the weird in-tarball directories for you. Then you don't need to adjust `S` at all.

**Comment by switch87** [comment]
@mgorny the license files are the ones included with the package. I added the newline.

**Comment by mgorny** [comment]
Hahah, we misunderstood for a bit ;-). The file was lacking newline character after the last line. You have added another empty line (so two newlines) ;-). Most of Linux editors add the missing newline automatically when you open and save the file. So please remove the extra empty line ;-D.

**Comment by mgorny** [comment]
@switch87, did you miss my notes on MY_PV?

**Comment by switch87** [comment]
It's the same way in all jet-brains packages, why change it all of a sudden?

**Comment by mgorny** [comment]
Because it sucks and increases maintenance burden? ;-)

**Comment by mgorny** [comment]
Nah, you still didn't get all of them. Remove two extra version components from filename. Kill MY_PN and versionator, just use `PhpStorm-${PV}` in SRC_URI. Remove S and inherit vcs-snapshot instead. That eclass will handle weird S value for you.

File: dev-util/phpstorm/phpstorm-9.0.2.ebuild [hunk 4]
```diff
@@ -0,0 +1,33 @@
+# Copyright 1999-2015 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Id$
+
+EAPI=5
+inherit eutils versionator vcs-snapshot
+
+SLOT="0"
```
Comment (mgorny) [inline dev-util/phpstorm/phpstorm-9.0.2.ebuild]: I'm sorry if I missed this before but why did you move `SLOT` here? It's recommended to use standard variable order to make ebuild reading easier for others, that is:

    DESCRIPTION
    HOMEPAGE
    SRC_URI

    LICENSE
    SLOT
    KEYWORDS
    IUSE

File: dev-util/phpstorm/phpstorm-9.0.2.ebuild [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+LICENSE="PhpStorm
+	|| ( PhpStorm_Academic PhpStorm_Classroom PhpStorm_OpenSource PhpStorm_personal )"
+IUSE=""
+KEYWORDS="~amd64 ~x86"
+
+RDEPEND=">=virtual/jdk-1.7:*"
+
+src_unpack() {
+	vcs-snapshot_src_unpack
+}
```
Comment (mgorny) [inline dev-util/phpstorm/phpstorm-9.0.2.ebuild]: vcs-snapshot is inherited last, so you can safely assume its `src_unpack()` will be called.

**Comment by mgorny** [comment]
@gentoo/proxy-maint, your approval? (consider the last issues are fixed)

File: dev-util/phpstorm/phpstorm-9.0.2.ebuild (hunk 5 above)
Comment (switch87) [inline dev-util/phpstorm/phpstorm-9.0.2.ebuild]: No it rails without this part

Comment (mgorny) [inline dev-util/phpstorm/phpstorm-9.0.2.ebuild]: Err, that's impossible. Either I or you are missing something…

**Comment by idella** [comment]
sorry just found email.  The DESCRIPTION appears to be around 100 chard long. Thought it was to come inside 80 chars excluding the var DESCRIPTION.

I have to check on the /opt location.

**Comment by mgorny** [comment]
@idella, thanks for noticing this. @switch87, please use a shorter `DESCRIPTION`. I think repoman requests around 68 chars. Also, please use `repoman full` ;-).

**Comment by idella** [comment]
oh sorry i edited already in the prior comment

**Comment by mgorny** [comment]
@idella:

1. Please don't edit comments like this, it is at least confusing.
2. As you can guess from the ebuild, it is a pre-built java package, so you could it a binary package. It's not built from sources.
3. The `-bin` suffix is used only to disambiguate when the same package is available in source and binary form. This is not the case here.

**Comment by idella** [comment]
ah ok. By chance it happened that you replied almost straight away, but as a rule I can go to a new comment and avoid possible confusion.
No source form available, ok that's fine. Leaves only the shortening of DESCRIPTION.

**Comment by mgorny** [comment]
@switch87, are you still there?
